tests/autorestart/test_container_autorestart.py

```diff
@@ -307,23 +307,34 @@ def check_all_critical_processes_status(duthost):
     return True
 
 
-def postcheck_critical_processes_status(duthost, container_autorestart_states, up_bgp_neighbors):
+def postcheck_critical_processes_status(duthost, feature_autorestart_states, up_bgp_neighbors):
     """Restarts the containers which hit the restart limitation. Then post checks
        to see whether all the critical processes are alive and
        expected BGP sessions are up after testing the autorestart feature.
 
     Args:
       duthost: An ansible object of DuT.
-      container_autorestart_states: A dictionary includes the container name (key) and
+      feature_autorestart_states: A dictionary includes the feature name (key) and
         its auto-restart state (value).
       up_bgp_neighbors: A list includes the IP of neighbors whose BGP session are up.
 
     Returns:
       True if post check succeeds; Otherwise False.
     """
-    for container_name in container_autorestart_states.keys():
-        if is_hiting_start_limit(duthost, container_name):
-            clear_failed_flag_and_restart(duthost, container_name)
+    for feature in feature_autorestart_states.keys():
+        if feature in duthost.DEFAULT_ASIC_SERVICES:
+            for asic in duthost.asics:
+                service_name = asic.get_service_name(feature)
+                container_name = asic.get_docker_name(feature)
+                if is_hiting_start_limit(duthost, service_name):
+                    clear_failed_flag_and_restart(duthost, service_name, container_name)
+        else:
+            # service_name and container_name will be same as feature
+            # name for features that are not in DEFAULT_ASIC_SERVICES.
+            service_name = feature
+            container_name = feature
```

Comment:
Since the `service_name` and `container_name` are same as `feature_name`, can we remove these two lines? Then the line 336 and 337 can be changed as following?

`if is_hiting_start_limit(duthost, feature_name):`

`clear_failed_flag_and_restart(duthost, feature_name, feature_name)`

Reply:
Removed those two lines.